src/main/java/apoc/mongodb/MongoDB.java

```diff
@@ -62,38 +66,62 @@ private Coll getColl(@Name("host") String hostOrKey, @Name("db") String db, @Nam
     }
 
     @Procedure
     @Description("apoc.mongodb.insert(host-or-port,db-or-null,collection-or-null,list-of-maps) - inserts the given documents into the mongodb collection")
-    public void insert(@Name("host") String hostOrKey, @Name("db") String db, @Name("collection") String collection, @Name("documents") List<Map<String,Object>> documents) {
-        getColl(hostOrKey, db, collection).insert(documents);
+    public void insert(@Name("host") String hostOrKey, @Name("db") String db, @Name("collection") String collection, @Name("documents") List<Map<String,Object>> documents) throws
+            MissingDependencyException {
+        getMongoColl(hostOrKey, db, collection).insert(documents);
     }
 
     @Procedure
     @Description("apoc.mongodb.delete(host-or-port,db-or-null,collection-or-null,list-of-maps) - inserts the given documents into the mongodb collection")
-    public Stream<LongResult> delete(@Name("host") String hostOrKey, @Name("db") String db, @Name("collection") String collection, @Name("query") Map<String,Object> query) {
-        return Stream.of(new LongResult(getColl(hostOrKey, db, collection).delete(query)));
+    public Stream<LongResult> delete(@Name("host") String hostOrKey, @Name("db") String db, @Name("collection") String collection, @Name("query") Map<String,Object> query) throws
+            MissingDependencyException {
+        return Stream.of(new LongResult(getMongoColl(hostOrKey, db, collection).delete(query)));
     }
     @Procedure
     @Description("apoc.mongodb.update(host-or-port,db-or-null,collection-or-null,list-of-maps) - inserts the given documents into the mongodb collection")
-    public Stream<LongResult> update(@Name("host") String hostOrKey, @Name("db") String db, @Name("collection") String collection, @Name("query") Map<String,Object> query, @Name("update") Map<String,Object> update) {
-        return Stream.of(new LongResult(getColl(hostOrKey, db, collection).update(query,update)));
+    public Stream<LongResult> update(@Name("host") String hostOrKey, @Name("db") String db, @Name("collection") String collection, @Name("query") Map<String,Object> query, @Name("update") Map<String,Object> update) throws
+            MissingDependencyException {
+        return Stream.of(new LongResult(getMongoColl(hostOrKey, db, collection).update(query,update)));
     }
 
     private String getMongoDBUrl(String hostOrKey) {
         return new UrlResolver("mongodb", "localhost", 27017).getUrl("mongodb", hostOrKey);
     }
 
+    private Coll getMongoColl(String hostOrKey, String db, String collection){
+        Coll coll = null;
+        try {
+            coll = getColl(hostOrKey, db, collection);
+        }
+        catch (NoClassDefFoundError e) {
+            throw new MissingDependencyException("Cannot find the jar into the plugins folder. \n"+
+                    "Please put these jar in the plugins folder :\n\n" +
+                    "bson-3.4.2.jar\n" +
```

Comment:
probably leave off the exact versions? and just do -.x.y.z and link to the docs.